packages/manager/modules/manager-pci-common/src/components/shape-input/ShapeInput.component.tsx

```diff
@@ -117,6 +117,9 @@ export const ShapesInputComponent = function ShapesInputComponent<T>({
     value,
     group: undefined,
   });
+  const [isAccordionOpen, setIsAccordionOpen] = useState<{
+    [groupName: string]: boolean;
```

Comment:
Could we use  instead an object with known properties to enhance typing ?
Because here I can add anything as a key

Reply:
the groupName can be any string so i'm not sure that it's possible to improve it, the component is pretty generic

Reply:
@frenautvh 
Why not creating an intersection of types ? 
I mean we can declare known keys as an union type and extend the object with opaque keys, what do you think?

Reply:
i'm not sure of the exact behavior of this component as i'm not the author of it, from what i understand the ShapeInput acceps items which can be grouped or stacked using key of type string

these keys are unknown and can be any string, so i don't see how you can use a more specific type than just "string" unless you have an example proving me wrong :)

Reply:
AFAIS, as we are using `none` property, it's a known one :)
So could be something like that :
```ts
type TFoo = { none: string } & { [key: string]: string };
```

Reply:
just added

```typescript
type TGroupName = 'none' | string;
```

please let me know if it's fine for you

Reply:
Mmh I will keep the original answer as it's not the same ;)

```ts
type TGroupName = 'none' | string;
```

This means that `TGroupName` can be any string, but TypeScript does not treat 'none' as a special case. In fact, 'none' | string is equivalent to just string because string already includes all possible string values.

Doing this: 

```ts
type Foo = { none: boolean } & { [key: string]: boolean };
```

Ensures that 'none' is required while still allowing other keys. :)

Reply:
fair enough i just updated this part